octodns/record/svcb.py

```diff
@@ -0,0 +1,251 @@
+from .chunked import _ChunkedValue
+from .rr import RrParseError
+
+SUPPORTED_PARAMS = {}
+
+
+def validate_svcparam_port(svcparamvalue):
+    reasons = []
+    try:
+        port = int(svcparamvalue)
+        if 0 < port > 65535:
+            reasons.append(f'port {port} is not a valid number')
+    except ValueError:
+        reasons.append('port is not a number')
+    return reasons
+
+
+def validate_svcparam_alpn(svcparamvalue):
+    reasons = []
+    alpns = svcparamvalue.split(',')
+    for alpn in alpns:
+        reasons += _ChunkedValue.validate(alpn, 'SVCB')
+    return reasons
+
+
+def validate_svcparam_iphint(ip_version, svcparamvalue):
+    reasons = []
+    addresses = svcparamvalue.split(',')
+    for address in addresses:
+        try:
+            if ip_version == 4:
+                IPv4Address(address)
+            if ip_version == 6:
+                IPv6Address(address)
+        except AddressValueError:
+            reasons.append(
+                f'ip{ip_version}hint "{address}" is not a valid IPv{ip_version} address'
+            )
+    return reasons
+
+
+def validate_svcparam_ip4hint(svcparamvalue):
+    return validate_svcparam_iphint(4, svcparamvalue)
+
+
+def validate_svcparam_ip6hint(svcparamvalue):
+    return validate_svcparam_iphint(6, svcparamvalue)
+
+
+def validate_svcparam_mandatory(svcparamvalue):
+    reasons = []
+    mandatories = svcparamvalue.split(',')
+    for mandatory in mandatories:
+        if (
+            mandatory not in SUPPORTED_PARAMS.keys()
+            and not mandatory.startswith('key')
+        ):
+            reasons.append(f'unsupported SvcParam "{mandatory}" in mandatory')
+        if mandatory.startswith('key'):
+            reasons += validate_svckey_number(mandatory)
+    return reasons
+
+
+def validate_svcparam_ech(svcparamvalue):
+    try:
+        b64decode(svcparamvalue, validate=True)
+    except binascii_error:
+        return ['ech SvcParam is invalid Base64']
+
+
+def validate_svckey_number(paramkey):
+    try:
+        paramkeynum = int(paramkey[3:])
+        if 7 < paramkeynum > 65535:
+            return [f'SvcParam key "{paramkey}" has wrong key number']
+    except ValueError:
+        return [f'SvcParam key "{paramkey}" has wrong format']
+    return []
+
+
+SUPPORTED_PARAMS = {
+    'no-default-alpn': {'has_arg': False},
+    'alpn': {'validate': validate_svcparam_alpn},
+    'port': {'validate': validate_svcparam_port},
+    'ipv4hint': {'validate': validate_svcparam_ip4hint},
+    'ipv6hint': {'validate': validate_svcparam_ip6hint},
+    'mandatory': {'validate': validate_svcparam_mandatory},
+    'ech': {'validate': validate_svcparam_ech},
+}
+
+
+class SvcbValue(EqualityTupleMixin, dict):
+
+    @classmethod
+    def parse_rdata_text(cls, value):
+        try:
+            # XXX: these are called SvcPriority, TargetName, and SvcParams in RFC 9460 section 2.
+            #       Should we mirror these names, or are priority, target and params good enough?
```

Comment:
Generally we've matched the names used in RFCs, e.g. MX does have `priority` & `target`. IMO It's annoying that they're prefixed/compound, but we should probably continue to match them. Everything else in octoDNS "should" be doing so and anything that doesn't is a "bug"